lib/impress.client.js

```diff
@@ -569,13 +573,15 @@ Client.prototype.end = function(
     }
     if (isString && length > impress.COMPRESS_ABOVE) {
       api.zlib.gzip(cache.data, (err, data) => {
-        if (err) client.error(500, err);
-        else if (!client.res.headersSent && client.res.setHeader) {
+        if (err) return client.error(500, err);
+        if (!client.res.headersSent && client.res.setHeader) {
```

Comment:
For the sake of consistency, `if (client.res.headersSent || !client.res.setHeader) return client.res.end();`